View/ChannelForm.cs

```diff
@@ -75,43 +92,49 @@ private void OnBrowserAddressChanged(object sender, AddressChangedEventArgs e)
             }
         }
 
+        private int retries = 0;
+        private DateTime LastRetry = DateTime.Now;
 
-        private void Channel_RaceChanged(object sender, EventArgs e)
+        private async void Channel_RaceChanged(object sender, EventArgs e)
         {
             try
             {
                 if (!IsDisposed)
                 {
                     Text = $"{Channel.Race.Goal} [{Channel.Race.GameName}] - {Channel.Race.ChannelName}";
 
-                    if (chatBox.IsBrowserInitialized == true)
+                    if (chatBox.Created == true)
                     {
-                        new Thread(() =>
+                        await Task.Delay(3000);
+
+                        if (retries <= 5)
                         {
-                            Thread.CurrentThread.IsBackground = true;
-                            System.Threading.Thread.Sleep(3000);
-                            if (retries <= 5)
+                            if (Channel.Token != null)
                             {
-                                if (Channel.Token != null)
-                                {
-                                    if (IsHandleCreated)
-                                    {
-                                        chatBox.BeginInvoke((Action)(() => chatBox.RequestHandler = new BearerAuthRequestHandler(Channel.Token.AccessToken)));
```

Comment:
I think I had left this here for an edge case that you could run into on first auth, we might not need it in webView2, but I might suggest if it dosen't hurt we leave it there.